### ISSUE TITLE
Changelog for release contains changes in their prereleases

### ISSUE THREAD

**Issue by dani8art** [webpro/release-it#416]
It would be good to have all the commit between the current release and the previous one in the changelog, I mean, If I'm not wrong, currently, if we have this:

```
$> git log 

* fix(): merge (tag 0.1.0)
* fix(): example 
* fix(): example 1 (tag 0.1.0-beta.2)
* fix(): example 2 
* fix(): example 3 (tag 0.1.0-beta.1)
* feat(): example4 (tag 0.0.5)
```
 We will generate these releases and changelogs on github:

### 0.1.0
* fix(): merge
* fix(): example 

### 0.1.0-beta.2
* fix(): example 1
* fix(): example 2

### 0.1.0-beta.1
* fix(): example 3

The idea is that we could have something like this:

### 0.1.0
* fix(): merge 
* fix(): example 
* fix(): example 1
* fix(): example 2 
* fix(): example 3

### 0.1.0-beta.2
* fix(): example 1
* fix(): example 2

### 0.1.0-beta.1
* fix(): example 3

Whit that, we will integrate all changes between 0.0.5 and 0.1.0 in the release changelog.
What do you think @webpro ?

**Comment by webpro** [comment]
This doesn't look common to me. Do you have any real-world examples?

However, you can override the `scripts.changelog` command any way you like. You can point it to e.g. a `bin/changelog.sh` script that outputs to `stdout`, and `release-it` will pick it up.

**Comment by dani8art** [comment]
Yeah! it could be a solution to my problem, but, would It be possible to have available the ${version} and the others variables? such as in `scripts.beforeStart`. 

https://github.com/webpro/release-it/blob/d6ea80a954b27bbb8f24ffadec0987fde3dc1c5c/lib/git.js#L116

Doing `format(...)` whit config too, right?

I have a solution to my problem but I would need ${version} in order to decide.
```
if [ "${version}" != *"next"* ];then
    git log --pretty=format:'* %s (%h)' $(git tag | grep -v next | sed '$!d')...HEAD | grep -v chore
else
    git log --pretty=format:'* %s (%h)' [REV_RANGE]
fi
```

**Comment by webpro** [comment]
You could do something like `"scripts.changelog": "script.sh ${version}"` and pick up `$1` in the script.

**Comment by dani8art** [comment]
I'm doing: 
```
"scripts": {
   "changelog": "echo ${version}"
}
```
```
  release-it:shell { command: 'echo ${version}',
  release-it:shell   options: { isReadOnly: true },
  release-it:shell   code: 0,
  release-it:shell   stdout: '',
  release-it:shell   stderr: '' } +7ms
```
But version is empty. I think changelogCommand is not formated: 
https://github.com/webpro/release-it/blob/d6ea80a954b27bbb8f24ffadec0987fde3dc1c5c/lib/git.js#L121

**Comment by webpro** [comment]
You're right, I'm sorry. Should be fixed in the next release.

This is basically the same:

```
"scripts": {
  "changelog": "script.sh `git describe --tags --abbrev=0`"
}
```

**Comment by dani8art** [comment]
`git describe --tags --abbrev=0` is the latest right? I would need the current version, the version is releasing at the moment.

**Comment by webpro** [comment]
Then you'll need to look up the command for that :-)

**Comment by webpro** [comment]
Provided you're not using `conventional:*` to bump, perhaps something like this:

```
{
  "scripts": {
    "afterBump": "script.sh ${version} > tmp-changelog",
    "release": "cat tmp-changelog"
  }
}
```

**Comment by webpro** [comment]
> But version is empty. I think changelogCommand is not formated:

This has been fixed in [v9.0.1](https://github.com/webpro/release-it/releases/tag/9.0.1).

Going to close this issue, as the initial question has been discussed. Feel free to discuss further, though.

**Comment by dani8art** [comment]
Cool! Thank you so much!
